packages/env/lib/config/config.js

```diff
@@ -155,23 +142,21 @@ module.exports = async function readConfig( configPath ) {
 
 	// Merge each of the specified environment-level overrides.
 	const allPorts = new Set(); // Keep track of unique ports for validation.
-	const env = allEnvs.reduce( ( result, environment ) => {
-		result[ environment ] = parseConfig(
+	const env = {};
+	for ( const envName in allEnvs ) {
```

Comment:
Moved away from `reduce` because it's a lot easier to do async/await in a plain for loop